azure-iot-device/azure/iot/device/common/transport_errors.py

```diff
@@ -0,0 +1,55 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module defines errors that may be raised from a transport"""
+
+
+class OperationCancelledError(Exception):
+    """
+    Operation was cancelled.
+    """
+
+    pass
+
+
+class ConnectionFailedError(Exception):
+    """
+    Connection failed to be established
+    """
+
+    pass
+
+
+class ConnectionDroppedError(Exception):
+    """
+    Previously established connection was dropped
+    """
+
+    pass
+
+
+class UnauthorizedError(Exception):
```

Comment:
Thinking about renaming this to `AuthorizationError` to distinguish it from the 401 related service-level error